ext/logging.c

```diff
@@ -1,97 +1,163 @@
 #include "logging.h"
+#include "sidecar.h"
 
-#include <stdatomic.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
 
 #include "configuration.h"
 #include <main/SAPI.h>
 
-static inline ddog_CharSlice dd_zend_string_to_CharSlice(zend_string *str) {
-    return (ddog_CharSlice){ .len = str->len, .ptr = str->val };
+
+static void dd_log_set_level(bool debug) {
+    bool once = runtime_config_first_init ? get_DD_TRACE_ONCE_LOGS() : get_global_DD_TRACE_ONCE_LOGS();
+    if (debug) {
+        if (strcmp("cli", sapi_module.name) != 0 && (runtime_config_first_init ? get_DD_TRACE_STARTUP_LOGS() : get_global_DD_TRACE_STARTUP_LOGS())) {
+            ddog_set_log_level(DDOG_CHARSLICE_C("debug"), once);
+        } else {
+            ddog_set_log_level(DDOG_CHARSLICE_C("debug,startup=error"), once);
+        }
+    } else {
+        ddog_set_log_level(dd_zend_string_to_CharSlice(runtime_config_first_init ? get_DD_TRACE_LOG_LEVEL() : get_global_DD_TRACE_LOG_LEVEL()), once);
+    }
+
 }
 
-atomic_uintptr_t php_ini_error_log;
+void ddtrace_log_ginit(void) {
+    dd_log_set_level(get_global_DD_TRACE_DEBUG());
+}
+
+atomic_int ddtrace_error_log_fd = -1;
+atomic_uintmax_t dd_error_log_fd_rotated = 0;
 
-void ddtrace_bgs_log_minit(void) { atomic_store(&php_ini_error_log, (uintptr_t)NULL); }
+void ddtrace_log_minit(void) {
+    if (ZSTR_LEN(get_global_DD_TRACE_LOG_FILE())) {
+        int fd = VCWD_OPEN_MODE(ZSTR_VAL(get_global_DD_TRACE_LOG_FILE()), O_CREAT | O_RDWR | O_APPEND, 0666);
```

Comment:
(Not necessarily related to this line) How annoying would it be to have a test using `DD_TRACE_LOG_FILE`?

Reply:
Not annoying :-)